src/clp_logging/handlers.py

```diff
@@ -68,6 +68,22 @@ def _init_timeinfo(fmt: Optional[str], tz: Optional[str]) -> Tuple[str, str]:
     return fmt, tz
 
 
+def _encode_log_event(msg: str, last_timestamp_ms: int) -> Tuple[bytearray, int]:
+    """
+    Encodes the log event with the input log message and reference timestamp.
+
+    :param msg: Input log message.
+    :param last_timestamp_ms: The timestamp of the last log event. Will be used
+        to calculate the timestamp delta.
+    :return: A tuple of the encoded log event and the associated timestamp.
+    """
+    timestamp_ms: int = floor(time.time() * 1000)
+    clp_msg: bytearray = FourByteEncoder.encode_message_and_timestamp_delta(
+        timestamp_ms - last_timestamp_ms, (msg + "\n").encode()
```

Comment:
Lets move the `+ "\n"` to CLPBaseHandler.emit. This will avoid making sure we need to add it in each `_write` path.

(It probably should have been there in the original version as well.)